base_changeset/static/src/js/list_backend.js

```diff
@@ -0,0 +1,159 @@
+                self._updateChangeset();
+            });
+            return res;
+        },
+
+        _updateChangeset: function() {
+            var self = this;
+            var state = this.model.get(this.handle);
+            this.model.getChangeset(state.model, false).then(function(changeset) {
+                self.renderer.renderChangesetPopovers(changeset);
+            });
+        },
+
+        applyChange: function(id) {
+            this.model.applyChange(id).then(this.reload.bind(this));
+        },
+
+        rejectChange: function(id) {
+            this.model.rejectChange(id).then(this.reload.bind(this));
+        },
+    });
+
+    ListRenderer.include({
+        /* eslint-disable no-unused-vars */
+        init: function(parent, state, params) {
+            this._super.apply(this, arguments);
+            this.lastChangeset = false;
+        },
+
+        _onToggleOptionalColumnDropdown: function(ev) {
+            this._super.apply(this, arguments);
+            // HACK to trigger the renderChangesetPopovers when user has added / removed an optional column
+            if (!$(ev.target).hasClass("o_optional_columns_dropdown_toggle")) {
+                this.renderChangesetPopovers(this.lastChangeset);
+            }
+        },
+
+        renderChangesetPopovers: function(changeset) {
+            var self = this;
+            this.lastChangeset = changeset;
+            // Split changes by record id first
+            var changesetById = {};
+            _.each(changeset, function(changes, fieldName) {
+                _.each(changes, function(change) {
+                    var resId = change.record_id;
+                    if (resId in changesetById) {
+                        if (fieldName in changesetById[resId]) {
+                            changesetById[resId][fieldName].push(change);
+                        } else {
+                            changesetById[resId][fieldName] = [change];
+                        }
+                    } else {
+                        changesetById[resId] = {};
+                        changesetById[resId][fieldName] = [change];
+                    }
+                });
+            });
+            // Render the popover in the correct row / column position
+            _.each(Object.values(changesetById), function(changesById) {
+                _.each(changesById, function(changes, fieldName) {
+                    var resId = changes[0].record_id.split(",")[1];
+                    var dataId = self.state.data.filter(function(element) {
+                        return element.res_id == resId;
+                    });
+                    if (dataId.length > 0) {
+                        var bodyRow = self.$el.find("[data-id='" + dataId[0].id + "']");
+                        var $tr = self.$el.find("thead").find("tr");
+                        var i = 0;
+                        var cellPos = 0;
+                        _.each($tr.children(), function(td) {
```

Comment:
You can change this to ` _.each($tr.children(), function(td, i) {` so you don't need to create the variable on line 87 (`var i = 0;`).

You can also use `_.findIndex` https://underscorejs.org/#findIndex